packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -4568,13 +4568,21 @@ abstract class ComponentElement extends Element {
   }
 }
 
+T unsafeCast<T>(Object any) => any; // ignore: return_of_invalid_type
+
 /// An [Element] that uses a [StatelessWidget] as its configuration.
 class StatelessElement extends ComponentElement {
```

Comment:
Possibly too breaking now, but I would guess it would be fairly straightforward to write this class hierarchy so no casts are needed on Widget at all, web or native.

Reply:
See [here](https://github.com/flutter/flutter/pull/56168) for a quick spike on this approach.